src/tools/compiletest/src/runtest.rs

```diff
@@ -128,7 +128,13 @@ pub fn run(config: Arc<Config>, testpaths: &TestPaths, revision: Option<&str>) {
         print!("\n\n");
     }
     debug!("running {:?}", testpaths.file.display());
-    let mut props = TestProps::from_file(&testpaths.file, revision, &config);
+    let mut props = {
+        let file = match config.mode {
+            Mode::RunMake => &testpaths.file.join("rmake.rs"),
+            _ => &testpaths.file,
+        };
+        TestProps::from_file(file, revision, &config)
+    };
```

Comment:
In the future I'd like to experiment with making the canonical name of a run-make test be the path of its `rmake.rs`, rather than the path of the enclosing directory. That should help eliminate a bunch of special cases for dealing with `Mode::RunMake`.

But when I tried, there were enough little complications that I didn't want to do so immediately in this PR.

Reply:
I have found that working with the `Testpaths` abstraction to be tricky, among other things.

Reply:
That sounds reasonable to me. Yeah, there are some quirks with a specific primary test file vs directory, but I haven't done an extensive survey on that yet.